compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -3277,16 +3225,18 @@ abstract class GenJSCode extends plugins.PluginComponent
       }
     }
 
-    /** Gen actual actual arguments to a primitive JS call
-     *  This handles repeated arguments (varargs) by turning them into
-     *  JS varargs, i.e., by expanding them into normal arguments.
+    /** Gen actual actual arguments to a primitive JS call.
+     *
+     *  * Repeated arguments (varargs) are expanded
+     *  * Default arguments are omitted or replaced by undefined
+     *  * All arguments are boxed
      *
-     *  Returns an only tree which is a JS array of the arguments. In most
-     *  cases, it will be a js.JSArrayConstr with the expanded arguments. It will
-     *  not if a Seq is passed to a varargs argument with the syntax seq: _*.
+     *  Repeated arguments that cannot be expanded at compile time (i.e., if a
+     *  Seq is passed to a varargs parameter with the syntax `seq: _*`) will be
+     *  wrapped in a [[js.JSSpread]] node to be expanded at runtime.
      */
     private def genPrimitiveJSArgs(sym: Symbol, args: List[Tree])(
-        implicit pos: Position): js.Tree = {
+        implicit pos: Position): List[js.Tree] = {
```

Comment:
In this method you are assuming that none of the non-last individual parameters result in a `JSSpread`. If this is not the case, we will produce bad IR. Maybe we should explicitly guard against that.